tools/target.py

```diff
@@ -53,19 +51,31 @@ def run(target, build, should_pull, **kwargs):
             != 0
         )
 
-        # If there are changes in tracked files in the docker folder when comparing with master,
-        # we want to build the image and we don't want to pull the image
-        if changed:
-            build = True
-            should_pull = False
-
-        if should_pull:
+        if pull:
             # Pull the image from dockerhub
             platform.pull(defaults.image, target)
 
-        # If user wants or needs to build it locally, do that
+        # If user wants or needs to build it locally, do that unless the build context is the same as origin/master
```

Comment:
Where does it check if the build context is the same as origin/master? I can't see where it does that

Reply:
https://github.com/NUbots/NUbots/blob/a85369bf060cb27c54dfd4476c6e24e109214a48/tools/target.py#L45-L52

Reply:
just above that in the file (in parts we didn't change), we run `git diff` to check if there are changes in the `docker` directory. If there are, the "build context" has been changed (and is "dirty")

Reply:
Okay thanks I see it now